resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -661,6 +665,33 @@ private[spark] class Client(
       }).toMap
     }).getOrElse(Map.empty[String, String])
   }
+
+  private def checkForUserJarsFilesExistence: Unit = {
```

Comment:
I think the trend has been for these kinds of precondition methods to pass the appropriate things to inspect as parameters, as opposed to inspecting the instance variables directly.  We also have been running the validation inside of `run()` as opposed to in the constructor. See `parseCustomLabels()` for example.